php/admin/class-migration.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * Handles migration to the new plugin.
+ *
+ * @package   Block_Lab
+ * @copyright Copyright(c) 2020, Block Lab
+ * @license   http://opensource.org/licenses/GPL-2.0 GNU General Public License, version 2 (GPL-2.0)
+ */
+
+namespace Block_Lab\Admin;
+
+use Block_Lab\Component_Abstract;
+
+/**
+ * Class Migration
+ */
+class Migration extends Component_Abstract {
+
+	/**
+	 * The AJAX action to dismiss the migration notice.
+	 *
+	 * @var string
+	 */
+	const NOTICE_AJAX_ACTION = 'bl_dismiss_migration_notice';
+
+	/**
+	 * The action of the migration notice nonce.
+	 *
+	 * @var string
+	 */
+	const NOTICE_NONCE_ACTION = 'bl-migration-nonce';
+
+	/**
+	 * The name of the migration notice nonce.
+	 *
+	 * @var string
+	 */
+	const NOTICE_NONCE_NAME = 'bl-migration-nonce-name';
+
+	/**
+	 * The slug of the stylesheet for the migration notice.
+	 *
+	 * @var string
+	 */
+	const NOTICE_STYLE_SLUG = 'block-lab-migration-notice-style';
+
+	/**
+	 * The slug of the script for the migration notice.
+	 *
+	 * @var string
+	 */
+	const NOTICE_SCRIPT_SLUG = 'block-lab-migration-notice-script';
+
+	/**
+	 * The user meta key to store whether a user has dismissed the migration notice.
+	 *
+	 * @var string
+	 */
+	const NOTICE_USER_META_KEY = 'block_lab_show_migration_notice';
+
+	/**
+	 * The user meta value stored if a user has dismissed the migration notice.
+	 *
+	 * @var string
+	 */
+	const NOTICE_DISMISSED_META_VALUE = 'dismissed';
+
+	/**
+	 * The capability required to see the notice.
+	 *
+	 * @var string
+	 */
+	const NOTICE_CAPABILITY = 'install_plugins';
+
+	/**
+	 * Adds an action for the notice.
+	 */
+	public function register_hooks() {
+		add_action( 'admin_notices', [ $this, 'render_migration_notice' ] );
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_assets' ] );
+		add_action( 'wp_ajax_' . self::NOTICE_AJAX_ACTION, [ $this, 'ajax_handler_migration_notice' ] );
+	}
+
+	/**
+	 * Outputs the migration notice if this is on the right page and the user has the right permission.
+	 */
+	public function render_migration_notice() {
+		if ( ! $this->should_display_migration_notice() ) {
+			return;
+		}
+
+		$migration_url = add_query_arg(
+			[
+				'post_type' => block_lab()->get_slug(),
+				'page'      => 'block-lab-migration',
+			],
+			admin_url( 'edit.php' )
+		);
+
+		?>
+		<div id="bl-migration-notice" class="notice notice-info bl-notice-migration">
+			<?php wp_nonce_field( self::NOTICE_NONCE_ACTION, self::NOTICE_NONCE_NAME, false ); ?>
+			<div class="bl-migration-copy">
+				<p>
+					<?php
+					printf(
+						/* translators: %1$s: the plugin name */
+						esc_html__( 'The Block Lab team have moved. For future updates and improvements, migrate now to the new home of custom blocks: %1$s.', 'block-lab' ),
+						'<strong>Genesis Custom Blocks</strong>'
```

Comment:
This isn't translated, as it's the name of the plugin. Maybe it should be 😄

Reply:
I think it _should_ be translatable, but the translation should be "Genesis Custom Blocks" in every language.

So, yeah, switch to a `__()` here.